warden/x/async/keeper/futures.go

```diff
@@ -0,0 +1,77 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	"cosmossdk.io/collections"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/warden-protocol/wardenprotocol/warden/repo"
+	types "github.com/warden-protocol/wardenprotocol/warden/x/async/types/v1beta1"
+)
+
+type FutureKeeper struct {
+	futures         repo.SeqCollection[types.Future]
+	futureByCreator collections.KeySet[collections.Pair[sdk.AccAddress, uint64]]
```

Comment:
How `futureByCreator` would be used? Answer because don't see queries to it. May be we can remove this if there will no use. I would expect map from creator to array of ids

Reply:
the idea is that there will be a query at some point, to fetch all your pending/completed futures. Maybe I should've wrote the query in this PR too, sorry 😅 

if that's okay I'd leave it as it is for now